src/lib.rs

```diff
@@ -144,17 +144,17 @@ mod tests {
         let todo = build_simple_operation();
 
         assert_eq!(
-            todo.to_operation_string(),
+            &todo.to_string(),
             "add\0ember-template-lint\0bare-strings\0some/path/here\00\00\00\05\0hello\01000\00\00"
         );
     }
 
     #[test]
     fn it_can_read_todo_operation_back_from_string() {
         let todo = build_simple_operation();
-        let s = todo.to_operation_string();
+        let s = todo.to_string();
 
-        assert_eq!(operation_from_string(s), todo);
+        assert_eq!(TodoOperation::from(s.as_str()), todo);
```

Comment:
This made me realize that what I *should* have implemented is `AsRef<str>`; I'll push an update. Then it will *just work* with `&s`.